staging/src/k8s.io/client-go/tools/remotecommand/remotecommand.go

```diff
@@ -140,3 +142,50 @@ func (e *streamExecutor) Stream(options StreamOptions) error {
 
 	return streamer.stream(conn)
 }
+
+// StreamContext opens a protocol streamer to the server and streams until a client closes
+// the connection or the server disconnects, but it can be closed through the Context.
+func (e *streamExecutor) StreamContext(ctx context.Context, options StreamOptions) error {
```

Comment:
Can you modify the existing method such that it makes a context and calls this method? That way there won't be so much code duplication and the only tests you'll have to write are those verifying that context cancellation works. Speaking of tests, this change needs some :)

Reply:
Thanks for your suggestion. Do I need to add Context to the `Stream` method instead of using `StreamContext`? If so, then I will modify the test.